frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -1235,6 +1235,23 @@ export const surveyLogic = kea<surveyLogicType>([
                 return null
             },
         ],
+        deviceTypesMatchTypeValidationError: [
+            (s) => [s.survey],
+            (survey: Survey): string | null => {
+                if (
+                    survey.conditions?.deviceTypesMatchType === SurveyMatchType.Regex &&
+                    survey.conditions.deviceTypes
+                ) {
+                    try {
+                        new RegExp(survey.conditions.deviceTypes?.join('|'))
+                    } catch (e: any) {
+                        return e.message
```

Comment:
i think we should return a human readable message here instead of an exception message

Reply:
![image](https://github.com/user-attachments/assets/bc992702-0284-4980-b90f-6a041cd0b9d1)

this is the message returned. I think it's readable enough